libnd4j/include/helpers/shape.h

```diff
@@ -1756,6 +1761,34 @@ INLINEDEF _CUDA_HD Nd4jLong coords2index(const Nd4jLong *shapeInfo, const Nd4jLo
     return index;
 }
 
+//////////////////////////////////////////////////////////////////////
+INLINEDEF _CUDA_HD Nd4jLong coords2index(const Nd4jLong *shapeInfo, const int *coords) {
+
+    Nd4jLong index, shift = 1;;
+
+    index = coords[shapeInfo[0] - 1];
+    for(uint i = shapeInfo[0]; i > 1; --i) {
+        shift *= shapeInfo[i];
+        index += shift * coords[i - 2];
+    }
+
+    return index;
+}
+
+//////////////////////////////////////////////////////////////////////
+INLINEDEF _CUDA_HD Nd4jLong coords2index(const Nd4jLong *shapeInfo, const uint *coords) {
+
+    Nd4jLong index, shift = 1;;
+
+    index = coords[shapeInfo[0] - 1];
+    for(uint i = shapeInfo[0]; i > 1; --i) {
```

Comment:
Please stop using direct indexing. Use shape::rank(..) instead. ShapeInfo format is going to change in foreseeable future